jerry-core/ecma/base/ecma-helpers-string.cpp

```diff
@@ -1111,29 +1115,9 @@ ecma_compare_ecma_strings_longpath (const ecma_string_t *string1_p, /* ecma-stri
 
         return ecma_compare_chars_collection (chars_collection1_p, chars_collection2_p);
       }
-      case ECMA_STRING_CONTAINER_LIT_TABLE:
-      {
-        JERRY_ASSERT (string1_p->u.lit_cp.packed_value != string2_p->u.lit_cp.packed_value);
-
-        return false;
-      }
-      case ECMA_STRING_CONTAINER_MAGIC_STRING:
+      default:
       {
-        JERRY_ASSERT (string1_p->u.magic_string_id != string2_p->u.magic_string_id);
-
-        return false;
-      }
-      case ECMA_STRING_CONTAINER_MAGIC_STRING_EX:
-      {
-        JERRY_ASSERT (string1_p->u.magic_string_ex_id != string2_p->u.magic_string_ex_id);
-
-        return false;
-      }
-      case ECMA_STRING_CONTAINER_UINT32_IN_DESC:
-      {
-        JERRY_ASSERT (string1_p->u.uint32_number != string2_p->u.uint32_number);
-
-        return false;
+        break;
```

Comment:
I did try  if/else , the performance is not good on ARM, even a little bit of regression. So I keep  switch.